assets/service_now_resource_setup_handler/index.py

```diff
@@ -80,83 +100,182 @@ def get_secret_value(self, secret_arn: str) -> Optional[str]:
+            bucket = self.__get_parameter(self.private_key_asset_bucket_param_name)
+            key = self.__get_parameter(self.private_key_asset_key_param_name)
+            
+            if not bucket or not key:
+                logger.error("Missing S3 bucket or key for private key asset")
+                return None
+            
+            # Read private key using S3 resource
+            s3_object = self.s3_resource.Object(bucket, key)
+            private_key = s3_object.get()['Body'].read().decode('utf-8')
+            
+            if not user_id or not client_id:
+                logger.error("Missing user ID or client ID for JWT")
+                return None
+            
+            # Create JWT payload
+            payload = {
+                "iss": client_id,
+                "sub": user_id,
+                "aud": client_id,
+                "iat": int(time.time()),
+                "exp": int(time.time()) + 3600,
+                "jti": str(uuid.uuid4())
             }
+            
+            # Encode JWT
+            encoded_jwt = jwt.encode(payload, private_key, algorithm="RS256", headers=JWT_HEADER)
+            return encoded_jwt
+            
         except Exception as e:
-            logger.error(f"Error getting request headers: {str(e)}")
+            logger.error(f"Error generating JWT: {str(e)}")
             return None
+        
+    def __get_jwt_oauth_access_token(self) -> Optional[str]:
+        """Get OAuth access token using JWT authentication.
 
-    def __get_request_base_url(self):
-        """Get base URL for ServiceNow API requests.
+        Returns:
+            Optional[str]: OAuth access token or None if retrieval fails
+        """
+        try:
+            if not self.instance_id:
+                logger.error("No ServiceNow instance id provided")
+                return None
+            
+            # Get parameters for JWT OAuth
+            client_secret = self.__get_parameter(self.client_secret_param_name)
+            client_id = self.__get_parameter(self.client_id_param_name)
+            user_id = self.__get_parameter(self.user_id_param_name)
+            
+            # Get encoded JWT
+            encoded_jwt = self.__get_encoded_jwt(client_id, user_id)
+
+            # Prepare request headers, data to get OAuth access token using encoded_jwt
+            token_url = self.__get_jwt_oauth_token_url()
+            headers = {
+                        'Content-Type': JWT_CONTENT_TYPE,
+                        'Authentication': f"Bearer {encoded_jwt}"
+                    }
+            data = {
+                'grant_type': JWT_GRANT_TYPE,
+                'assertion': encoded_jwt,
+                'client_id': client_id,
+                'client_secret': client_secret
+            }
+            response = requests.post(token_url, headers=headers, data=data)
+            return (response.json())['access_token']
+        except Exception as e:
```

Comment:
issue(blocking): Because the except is so broad, there's a chance we leak the token.  Take a look at [requests.exceptions documentation](https://requests.readthedocs.io/en/latest/_modules/requests/exceptions/).